### DISCUSSION TITLE
Husky for non npm projects

### DISCUSSION THREAD

**Discussion by maxiride** [typicode/husky#1094] [Q&A] [answered]
Is it possible to use Husky for non node\npm projects in other languages? If so, it hasn't been clear to me on how to proceed and would really appreciate some guidance.

I've already configured semantic-release for non npm projects since it can run globally with npx (I am fine with having node installed but not having the node_modules folder in the project nor a package.json) and I would like to validate commits before they are created on the git history, possibly validating pull requests too.

**Comment by typicode** [comment, marked as answer]
I'd say no.

However if you're fine with using npx for global install, you can do `npx husky install && npx husky add .husky/pre-commit 'echo foo'`.
The downside of it is that if you're working in a team, people won't have husky installed unless they also run `npx husky install`.

**Reply by maxiride** [reply]
Thanks for the clarifications and the tip

**Reply by typicode** [reply]
You're welcome

**Reply by jdlopez-fe** [reply]
In the new version of husky, the command install and add are Deprecated, ho to proceed in that case?

**Comment by VasylHerman** [comment]
I use one simple line for that, however again, there is no guarantee 

`echo "[ -d .git ] && npx husky install > /dev/null || true" >> ~/.zshrc`

https://gist.github.com/PrettySolution/ba76733e0d963481130806433b07eaf1
